sources/lib/DependencyInjection/PommExtension.php

```diff
@@ -60,11 +62,42 @@ public function configure(array $configs, ContainerBuilder $container)
         $configuration = new Configuration();
         $config = $processor->processConfiguration($configuration, $configs);
 
-        $container->setParameter('pomm.configuration', $config['configuration']);
+        $definition = $container->getDefinition('pomm');
+        foreach ($config['configuration'] as $name => $pommConfig) {
+            if (isset($pommConfig['service']) && isset($pommConfig['class:session_builder'])) {
+                throw new \RuntimeException(sprintf('Please configure either "service" or "class:session_builder" parameter on %s', $name));
+            } else if (isset($pommConfig['service'])) {
+                $service = $container->getDefinition($pommConfig['service']);
+                $service->setArguments([$pommConfig]);
+
+                $definition->addMethodCall('addBuilder', [$name, new Reference($pommConfig['service'])]);
+            } else {
+                $service = uniqid($pommConfig['class:session_builder'], true);
+                $cbDefinition = $container->register($service, ltrim($pommConfig['class:session_builder'], '\\'));
+                $cbDefinition->setShared(false);
+                $cbDefinition->setArguments([$pommConfig]);
+
+                $definition->addMethodCall('addBuilder', [$name, new Reference($service)]);
+                trigger_error('Using class:session_builder is deprecated since 2.3 and is going to be removed in 3.0 Use service and service id instead.', E_USER_DEPRECATED);
```

Comment:
missing ``@``.

And this should be triggered in the Configuration class as well (in a ``beforeNormalization()``), to be sure it is trigger each time the setting is used explicitly in the config file (ensuring that the Configuration class can remove this setting in 3.0). See https://github.com/symfony/symfony/blob/v2.8.15/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L293 for an example